File: src/disana.f
      subroutine disana(dis,tmin,tave,tmax,ncol)
c
      double precision dis(ncol,ncol)
      double precision tmin(ncol)
      double precision tave(ncol)
      double precision tmax(ncol)
      integer ncol
c
      do 10 i=1,ncol
      tmin(i) = 1.0
      tave(i) = 0.0
      tmax(i) = 0.0
        do 11 j=1,ncol
        if (i .eq. j) goto 11
        tmin(i) = min(tmin(i),dis(i,j))
        tave(i) = tave(i) + dis(i,j)
        tmax(i) = max(tmax(i),dis(i,j))
   11   continue
      tave(i) = tave(i) / (ncol-1)
   10 continue
c
      end

File: R/disana.R
disana <- function (x)
{
    if (class(x) == "dist") {
        y <- as.matrix(x)
        triang <- x
    }
    else {
        y <- as.matrix(x)
        triang <- y[row(y) > col(y)]
    }
    is.na(diag(y)) <- TRUE 
    tmin <- apply(y,1,function(z){min(z,na.rm=TRUE)})
    tavg <- apply(y,1,function(z){mean(z,na.rm=TRUE)})
    tmax <- apply(y,1,function(z){max(z,na.rm=TRUE)})
    plots <- NULL
    plot(sort(triang), xlab = "Sorted Value", ylab = "Dissimilarity")
    readline("Press return for next page....")
    plot(sort(tmin), ylim = c(0, max(tmax)), xlab = "Sorted Plot",
            ylab = "Dissimilarity")
    points(sort(tavg), col = 2)
    points(sort(tmax), col = 3)
    readline("Press return for next page....")
    plot(tmin, tavg, xlab = "Minimum Dissimilarity",
            ylab = "Average Dissimilarity")
    lines(c(0.5, 0.5), c(min(tavg), max(tavg)), col = 2)
    yorn <- readline("Do you want to identify individual plots [Y or N] : ")
    if (yorn == 'Y' || yorn == 'y') 
               plots <- identify(tmin,tavg,attr(x,'Labels'))
    res <- list(min = tmin, mean = tavg, max = tmax, plots = plots)
    invisible(res)
}
